lib/riemann/helpers/event.ex

```diff
@@ -59,7 +40,46 @@ defmodule Riemann.Helpers.Event do
         |> Map.delete(:metric_sint64)
         |> Map.put(:attributes, attributes)
       end
+    end
+  end
+
+  @moduledoc false
+  def build(args, mod) do
+    args
+    |> Enum.into(%{})
+    |> Map.put_new(:time, :erlang.system_time(:seconds))
+    |> Map.put_new_lazy(:host, &default_event_host/0)
+    |> set_attributes_field
+    |> set_metric_pb_fields
+    |> Map.to_list
+    |> mod.new()
+  end
 
+  defp default_event_host do
+    event_host_setting() || machine_hostname()
+  end
+
+  defp set_attributes_field(map) do
+    case Map.get(map, :attributes) do
+      nil -> map
+      a   -> Map.put(map, :attributes, Attribute.build(a))
+    end
+  end
+
+  defp set_metric_pb_fields(map) do
```

Comment:
what do you think of splitting this up into a function match instead?

`defp set_metric_pb_fields(%{metric: i} = map) when is_integer(i) do` etc...